optuna/samplers/_tpe/sampler.py

```diff
@@ -839,8 +475,8 @@ def after_trial(
 
 
 def _get_observation_pairs(
```

Comment:
Thanks for the comment. As you said, we need to update the docs. 

> Missing parameters are not set to None but now skipped. I think this also changes the results of the below/above split

Oh, I didn't notice it, but you are right that the logic of split is a bit different when `group=True` and there is a conditional parameter. With this PR change, `below` will always contain gamma samples. I think this change will rather work in the direction of improving the estimation performance of the Parzen estimator for `below`.

However, it would be better to see the benchmark results when `group=True` and conditional parameters are present. In any case, since we have adapted the logic to multivariate TPE and we know that multivariate TPE achieves good performance in this situation, I think such benchmark results are good for future work.

Reply:
Yes, we need to update the docs.

> Missing parameters are not set to None but now skipped. I think this also changes the results of the below/above split

I didn't notice it, but you are right that the logic of split is a bit different when group=True and there is a conditional parameter. With this PR change, `below` will always contain gamma samples. I think this change will rather work in the direction of improving the estimation performance of the Parzen estimator for `below`.

However, it would be better to see the benchmark results when group=True and conditional parameters are present. In any case, since we have adapted the logic to multivariate TPE and we know that multivariate TPE achieves good performance in this situation, I think such benchmark results are good for future work.